mmv1/third_party/terraform/tests/resource_container_cluster_test.go.erb

```diff
@@ -4889,6 +4944,50 @@ resource "google_container_cluster" "primary" {
 `, name)
 }
 
+func testAccContainerCluster_withMonitoringConfigEnabled(name string) string {
+       return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  monitoring_config {
+      enable_components = [ "SYSTEM_COMPONENTS" ]
+  }
+}
+`, name)
+}
+
+<% if version == 'beta' -%>
+func testAccContainerCluster_withMonitoringConfigUpdated(name string) string {
+       return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  monitoring_config {
+         enable_components = [ "SYSTEM_COMPONENTS", "WORKLOADS" ]
+  }
+}
+`, name)
+}
+
+func testAccContainerCluster_withMonitoringConfigPrometheusUpdated(name string) string {
+       return fmt.Sprintf(`
+resource "google_container_cluster" "primary" {
+  name               = "%s"
+  location           = "us-central1-a"
+  initial_node_count = 1
+  monitoring_config {
+         enable_components = [ "SYSTEM_COMPONENTS", "WORKLOADS" ]
```

Comment:
Since one of the changes is to make this optional - would it be possible to remove it as a field in this config?

Reply:
Already implicitly done at line 1895 when we restore to testAccContainerCluster_basic at the end of  TestAccContainerCluster_withMonitoringConfig.

Reply:
That tests that monitoring_config as a whole is optional; having enable_components be unset in this test would ensure that it is optional even if a monitoring_config is provided. This is not necessarily a blocker, but it would be nice unless there's a particular reason this needs to be set.

Reply:
Done.